### PR TITLE
Add clang-tidy `performance-*` checks

### LINKED ISSUE #2504: Enable clang-tidy `performance-*` checks
## Subject of the issue

https://clang.llvm.org/extra/clang-tidy/checks/list.html

These checks do a good job finding simple ways we're writing inefficient code. I've used this locally and there aren't too many checks to disable before everything passes.

### PR DESCRIPTION
**Pull request by ChrisThrasher**
## Description

Closes #2504

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
Homebrew is acting up so macOS CI jobs are failing but I can confirm it works locally.

**Comment by ChrisThrasher** [comment]
https://github.com/actions/setup-python/issues/577

Others have had similar issues with Homebrew

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
       if: runner.os == 'macOS'
       run: |
         brew update
-        brew install gcovr ninja
+        brew install gcovr ninja || true
```
Comment (Bromeon) [inline .github/workflows/ci.yml:88]: If these installations fail, is it safe to just continue?

If yes, why do we install them in the first place?

**Review by Bromeon — COMMENTED** [review]
Great addition!

Just a small comment unrelated to the actual clang-tidy checks.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ChrisThrasher) [reply .github/workflows/ci.yml:88]: https://github.com/SFML/SFML/pull/2499#issuecomment-1500960157

It's a workaround for this Homebrew issue that appeared out of nowhere. It still installs the packages we care about so we can't remove `brew install`.

**Review by eXpl0it3r — APPROVED** [review]

**Pull request merged**